app/models/crop.rb

```diff
@@ -354,6 +335,23 @@ def self.search(query)
     end
   end
 
+  private
+
+  def add_names_to_list(names_to_add, list_name, col_name)
+    cropbot = Member.find_by_login_name('cropbot')
+    names_to_add.each do |n|
+      if self.send("#{list_name}_names").exists?(["#{col_name} LIKE ?", n])
```

Comment:
It could do with a comment or two to explain it clearly, took me a bit to get my head around the whole method.

Only tweaks I can really think of is stuff like:
```
actual_names_to_add = names_to_add.select {|name| !self.send("#{list_name}_names").exists?(["#{col_name} LIKE ?", name]) }
```
... which kills off one conditional.

Maybe the SQL could be tweaked to be ```SELECT DISTINCT column_name FROM list_name WHERE LOWER(column_name) IN(?)``` and pass in lowercase input; so you do one query for all existing instead of multiple.
Then just ```names_to_add - existing_names_query```